### PR TITLE
Fixes for GRU/LSTM in Vivado backend

### PR DESCRIPTION
**Pull request by dsrankin**
Fixes for GRU/LSTM in Vivado backend

This PR fixes issues noted in https://github.com/fastmachinelearning/hls4ml/issues/594 with the GRU/LSTM in the Vivado backend. 

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)

## Tests

Tested using the configuration in the above issue and confirmed to fix issues.

@bo3z

### CODE REVIEW

File: hls4ml/backends/vivado/vivado_backend.py [hunk 1]
```diff
@@ -267,8 +267,8 @@ def init_gru(self, layer):
         reuse_factor = layer.model.config.get_reuse_factor(layer)
         layer.set_attr('recurrent_reuse_factor', reuse_factor)
 
-        recurrent_bias = np.zeros(layer.weights['recurrent_weight'].shape[1])
-        layer.add_weights_variable(name='recurrent_bias', var_name='br{index}', data=recurrent_bias)
+        #recurrent_bias = np.zeros(layer.weights['recurrent_weight'].shape[1])
```
Comment (vloncar) [inline hls4ml/backends/vivado/vivado_backend.py:270]: Why don't we remove these lines?

File: test/pytest/test_rnn.py [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
-import tensorflow as tf
 import numpy as np
 from pathlib import Path
-from tensorflow.keras import optimizers
-from tensorflow.keras.models import Model
-from tensorflow.keras.layers import Input, Embedding, SimpleRNN, LSTM, GRU
-import math
-from tensorflow.keras import backend as K
+
+from traitlets import default
```
Comment (vloncar) [inline test/pytest/test_rnn.py:6]: Auto-import issue?

**Review by vloncar — CHANGES_REQUESTED** [review]
Looks good, minor cleanup should be done.

File: hls4ml/backends/vivado/vivado_backend.py (hunk 1 above)
Comment (drankincms) [reply hls4ml/backends/vivado/vivado_backend.py:270]: Whoops, good catch

File: test/pytest/test_rnn.py (hunk 2 above)
Comment (drankincms) [reply test/pytest/test_rnn.py:6]: Yeah, not sure where these came from... Reverting

**Pull request merged**